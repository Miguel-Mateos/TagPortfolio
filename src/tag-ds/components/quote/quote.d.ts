import React from 'react';
export interface IQuote {
    quote: string;
    /**
     * For setting the Avatar info
     */
    avatar?: {
        img?: string;
        title?: string;
        subtitle?: string;
    };
    /**
     * Additional styling if needed
     */
    className?: string;
    [others: string]: any;
}
export declare const Quote: React.FC<IQuote>;
export default Quote;
