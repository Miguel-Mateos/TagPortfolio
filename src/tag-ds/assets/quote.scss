.tag-ds {
  .quote-wrapper,
  &.quote-wrapper {
    width: calc(42 * var(--unit));
  }
  .quote {
    @extend .h3;
    padding-bottom: calc(3 * var(--unit));
  }
  .quote-icon {
    display: block;
    color: var(--accent500);
    font-size: calc(6 * var(--unit));
  }
  .avatar-title {
    @include font(base);
  }
  .avatar-subtitle {
    @include font(small);
  }
}

