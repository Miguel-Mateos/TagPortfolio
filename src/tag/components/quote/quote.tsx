import React from 'react';

export interface IQuote {
  quote: string;

  /**
   * For setting the Avatar info
   */
  avatar?: {
    img?: string;
    title?: string;
    subtitle?: string;
  };
  /**
   * Additional styling if needed
   */
  className?: string;
  [others: string]: any;
}

export const Quote: React.FC<IQuote> = (props) => {
  const {avatar, className, ...rest} = props;

  const content = (
    <div className={`tag-ds quote-wrapper ${className || ''}`} {...rest}>
      <span className="material-icons quote-icon">format_quote</span>
      <div className="quote">{props.quote}</div>
      {avatar && (
        <div className={'avatar-wrapper'}>
          {avatar.img && <img src={avatar.img} alt="avatar icon" className="avatar_xsmall" />}
          <div className="avatar-text-wrapper">
            {avatar.title && <span className="avatar-title">{avatar.title}</span>}
            {avatar.subtitle && <span className="avatar-subtitle">{avatar.subtitle}</span>}
          </div>
        </div>
      )}
    </div>
  );
  return content;
};

export default Quote;
