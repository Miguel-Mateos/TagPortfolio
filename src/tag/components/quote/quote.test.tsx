import React from 'react';
import { IQuote, Quote } from './quote';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

const QuoteTest = (args: IQuote) => {
    return (
        <Quote {...args} data-testid='quote'/>
    );
};

const avatar = {
    title: "Name and Surname",
    subtitle: "Charge",
    img: "https://www.w3schools.com/howto/img_avatar.png",
}

test('render quote component and match snapshot', () => {
    const component = renderer.create(<QuoteTest quote="Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa commodi officiis optio magni! Perferendis doloremque similique quas magnam repellendus facere voluptatem tenetur possimus cupiditate rerum suscipit, accusamus ullam magni odio."/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('render default quote component', () => {
    const { getByTestId } = render(<QuoteTest quote="Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa commodi officiis optio magni! Perferendis doloremque similique quas magnam repellendus facere voluptatem tenetur possimus cupiditate rerum suscipit, accusamus ullam magni odio."/>);
    const quote = getByTestId('quote');
    expect(quote).toBeInTheDocument();
});

test('render quote component with avatar', () => {
    const {container} = render(<QuoteTest avatar={avatar} quote="Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa commodi officiis optio magni! Perferendis doloremque similique quas magnam repellendus facere voluptatem tenetur possimus cupiditate rerum suscipit, accusamus ullam magni odio."/>);
    const quote = container.getElementsByClassName('avatar-wrapper')[0];
    expect(quote).toBeInTheDocument();
});

