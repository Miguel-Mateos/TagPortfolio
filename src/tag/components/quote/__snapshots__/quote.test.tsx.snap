// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render quote component and match snapshot 1`] = `
<div
  className="tag-ds quote-wrapper "
  data-testid="quote"
  quote="Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa commodi officiis optio magni! Perferendis doloremque similique quas magnam repellendus facere voluptatem tenetur possimus cupiditate rerum suscipit, accusamus ullam magni odio."
>
  <span
    className="material-icons quote-icon"
  >
    format_quote
  </span>
  <div
    className="quote"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa commodi officiis optio magni! Perferendis doloremque similique quas magnam repellendus facere voluptatem tenetur possimus cupiditate rerum suscipit, accusamus ullam magni odio.
  </div>
</div>
`;
